lib/flow/VaporField.cpp

```diff
@@ -156,18 +174,28 @@ VaporField::GetVelocity( float time, const glm::vec3& pos, glm::vec3& velocity,
         if( !InsideVolumeVelocity( time, pos ) )
             return OUT_OF_FIELD; 
 
-    // Retrieve the missing value velocity multiplier 
+    // Retrieve the missing value and velocity multiplier 
     const float mult = _params->GetVelocityMultiplier();
-    glm::vec3 missingV;
+    glm::vec3 missingV; // stores missing values for 3 velocity variables
+
+    // Create a constant grid in case it will be used for empty variables.
+    const VAPoR::ConstantGrid cgrid( 0.0f );
 
     if( IsSteady )
     {
         size_t currentTS = _params->GetCurrentTimestep();
         for( int i = 0; i < 3; i++ )
         {
-            auto& varname = VelocityNames[i];
-            grid = _getAGrid( currentTS, varname );
-            VAssert( grid );
+            const auto& varname = VelocityNames[i];
+            if( varname.empty() )
+            {
+                grid = &cgrid;
+            }
+            else
+            {
+                grid = _getAGrid( currentTS, varname );
```

Comment:
How about having _getAGrid return a constant grid as needed and eliminating all of these conditionals?

Reply:
fixed.